src/sqlfluff/core/dialects/dialect_ansi.py

```diff
@@ -1110,7 +1102,11 @@ def get_eventual_aliases(self) -> List[Tuple[BaseSegment, AliasInfo]]:
         """
         buff = []
         direct_table_children = self.get_children("table_expression")
-        join_clauses = self.get_children("join_clause")
+        join_clauses = [
+            join_clause
+            for child in direct_table_children
+            for join_clause in child.get_children("join_clause")
+        ]
```

Comment:
Now that join clauses are nested underneath `table_expression` segments this had to be tweaked slightly